### PR TITLE
New instructions semantics

### PR DESCRIPTION
**Pull request by apach301**

### CODE REVIEW

File: src/libtriton/arch/x86/x86Semantics.cpp [hunk 1]
```diff
@@ -7530,6 +7570,38 @@ namespace triton {
...
+        if (src.getType() == OP_REG) {
+          node = this->astCtxt->extract(triton::bitsize::dword-1, 0, node);
+          if (dst.getType() == OP_REG) {
+            auto op1 = this->symbolicEngine->getOperandAst(inst, dst);
+            auto upper = this->astCtxt->extract(triton::bitsize::dqword-1, triton::bitsize::dword, op1);
+            node = this->astCtxt->concat(upper, node);
+          }
+        }
+        else {
+          node = this->astCtxt->concat(this->astCtxt->bv(0, triton::bitsize::dqword - triton::bitsize::dword), node);
```
Comment (SweetVishnya) [inline src/libtriton/arch/x86/x86Semantics.cpp:7591]: No need to concat with zero.

SymbolicEngine::createSymbolicRegisterExpression -> SymbolicEngine::insertSubRegisterInParent will perform zero extension.

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: src/libtriton/arch/x86/x86Semantics.cpp (hunk 1 above)
Comment (apach301) [reply src/libtriton/arch/x86/x86Semantics.cpp]: Removed.

**Comment by JonathanSalwan** [comment]
Thanks for this pr :). let me few days to review it.

**Comment by JonathanSalwan** [comment]
Thanks for this PR. Can you add some tests in the [x86 semantics tester](https://github.com/JonathanSalwan/Triton/blob/dev-v0.9/src/testers/x86/unicorn_test_x86.py)?

**Comment by apach301** [comment]
Add test for one instruction. Tests on other instructions failed with unicorn errors (Invalid memory read (UC_ERR_READ_UNMAPPED), Invalid instruction (UC_ERR_INSN_INVALID)).

**Pull request merged**